### PR TITLE
Fixing getUri to return url that would be used for a request

### PR DESCRIPTION
**Pull request by tskimmett**
The original implementation of `Axios.prototype.getUri` did not take into consideration the `baseURL` of the instance configuration. Therefore, it would always return a relative URL instead of the URL that would be used if you actually made a request with the provided config.

I also added a test that validates this requirement.

### CODE REVIEW

File: lib/core/Axios.js [hunk 1]
```diff
@@ -58,6 +60,11 @@ Axios.prototype.request = function request(config) {
 
 Axios.prototype.getUri = function getUri(config) {
   config = mergeConfig(this.defaults, config);
+
+  if (config.baseURL && !isAbsoluteURL(config.url)) {
+    config.url = combineURLs(config.baseURL, config.url);
```
Comment (avindra) [inline lib/core/Axios.js]: This mutates the configuration object,  which will cause issues with subsequent requests. Any mutations to `url`  should be done in a local variable. e.g.,

```
let url = config.url;
if ( .... ) {
   url = combineURLs( ... );
}

return buildURL(url, ...);
```

Comment (tskimmett) [reply lib/core/Axios.js]: ‘mergeConfig’ returns a new configuration object, which means the config passed to ‘getUri’ is not mutated when we set the url.

**Comment by tskimmett** [comment]
Does anyone manage this repo?

**Comment by jasonsaayman** [comment]
@tskimmett the repo is managed by @emilyemorehouse I am trying to help out as much as I can, at the moment I cannot merge a pull request, however, I am able to test and review them and test them. I will try to test this one asap.

File: lib/core/Axios.js (hunk 1 above)
Comment (avindra) [reply lib/core/Axios.js]: Yikes. `mergeConfig` itself sounds like a big piece of tech debt that needs to go. It should be possible to structure this library such that we dont have to copy a fat object like that so many times. It's frequently a source of bugs, as experienced with `retry-axios` ( https://github.com/JustinBeckwith/retry-axios/issues/1  https://github.com/JustinBeckwith/retry-axios/issues/50  https://github.com/JustinBeckwith/retry-axios/issues/4  ) cc @emilyemorehouse Curious to hear your thoughts around `mergeConfig`

**Comment by smesko85** [comment]
Any updates on this?

**Comment by tskimmett** [comment]
This is waiting on the repo maintainers as far as I'm concerned.

**Comment by jasonsaayman** [comment]
Hi

I agree with chinesedfan sentiments in this pull request. Closing for now.

THanks

**Comment by namoojaalallee** [comment]
Good

**Pull request closed without merging**